### PR TITLE
CIV-8169 sign settlement agreement submit function notifications

### PR DESCRIPTION
**Pull request by jeswanth-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-8169

### Change description ###

sign settlement agreement submit function notifications

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by ElifDenizEsea — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -370,7 +370,7 @@ configurations.all {
 
 dependencies {
 
-  implementation 'com.github.hmcts:civil-commons:1.0.36'
+  implementation 'com.github.hmcts:civil-commons:v8169'
```
Comment (jeswanth-hmcts) [inline build.gradle:373]: this change will be removed once commons change is merged

**Review by portizdev — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantSignSettlementAgreementNotificationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import uk.gov.hmcts.reform.civil.model.citizenui.CaseDataLiP;
+import uk.gov.hmcts.reform.civil.notify.NotificationService;
+import uk.gov.hmcts.reform.civil.notify.NotificationsProperties;
+
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
```
[the remark this answers is not in the document]
Comment (jeswanth-hmcts) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantSignSettlementAgreementNotificationHandlerTest.java:27]: could you help me with example

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantSignSettlementAgreementNotificationHandler.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            FRONTEND_URL, pipInPostConfiguration.getCuiFrontEndUrl()
+        );
+    }
+
+    private CallbackResponse notifyForSignedSettlement(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        Map<String, String> templateProperties = addProperties(caseData);
+        String recipientEmail = isRespondentNotification(callbackParams) ? getRecipientEmailForRespondent(caseData) :
+            caseData.getApplicant1Email();
+        String templateId = isRespondentNotification(callbackParams) ? getTemplateID(
```
[the remark this answers is not in the document]
Comment (jeswanth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantSignSettlementAgreementNotificationHandler.java:75]: any suggestable good solution ?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantSignSettlementAgreementNotificationHandler.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    private boolean isRespondentNotification(CallbackParams callbackParams) {
+        return callbackParams.getRequest().getEventId()
+            .equals(NOTIFY_LIP_RESPONDENT_FOR_SIGN_SETTLEMENT_AGREEMENT.name());
+    }
+
+    private String getTemplateID(CaseData caseData, String agreedTemplate, String notAgreedTemplate) {
+        Optional<CaseDataLiP> optionalCaseDataLiP = Optional.ofNullable(caseData.getCaseDataLiP());
+        boolean isAgreed = optionalCaseDataLiP.map(CaseDataLiP::isDefendantSignedSettlementAgreement).orElse(false);
+        boolean isNotAgreed = optionalCaseDataLiP.map(CaseDataLiP::isDefendantSignedSettlementNotAgreed).orElse(false);
+        return isAgreed ? agreedTemplate : (isNotAgreed ? notAgreedTemplate : null);
```
[the remark this answers is not in the document]
Comment (jeswanth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantSignSettlementAgreementNotificationHandler.java]: simplified the condition

**Review by kenneth-hmcts — DISMISSED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Review by jarekPierchala — DISMISSED** [review]

**Pull request merged**